common/src/primitives/id.rs

```diff
@@ -29,6 +30,21 @@ impl From<GenericArray<u8, typenum::U32>> for H256 {
     }
 }
 
+impl H256 {
+    fn reverse_inner(&self) -> [u8; 32] {
+        let mut h256_inner = self.0;
+        h256_inner.reverse();
+        h256_inner
+    }
+}
+
+impl From<H256> for Uint256 {
+    fn from(hash: H256) -> Self {
+        let x = hash.reverse_inner();
```

Comment:
This isn't an operation I'm particularly happy about and I think we should discuss the reason why this is necessary. Since we're starting from scratch, let's try to make both H256 and Uint256 use the same order. No need to reverse things every time. This can get very confusing and error-prone very quickly.

Reply:
I can see that H256 uses big-endian order... we may have to fix this...

Reply:
As we briefly discussed earlier I think forking fixed_hash is the best way to approach this, my 30s peak makes it look like we should be dandy dependency wise so it's just a case of fixing the macros and tests and we should be good to go.